### ISSUE TITLE
Question about ggcoxadjustedcurves

### ISSUE THREAD

**Issue by kassambara** [kassambara/survminer#229]
Comment from a STHDA visitor (http://www.sthda.com/english/wiki/survminer-0-3-0)

@pbiecek, what do you think about the comment below?

Hi,

Keep up the good work! I wanted to ask a question regarding the function ggadjustedcurve(). From my understanding, this function works by predicting survival probabilties of each individual and then these survival probabilities are averaged by strata (ex. sex). I do not believe this is what is referenced by the "Terry M Therneau (2015); Adjusted survival curves" article. The way it is currently done results in different covariate distributions for each strata. For example, females will have a set of covariate values unique to them, whereas males will have a set of covariate values unique to them - I do not believe this is "adjusted". Ideally, we'd want both covariate distribution to be identical and the only factor changing would be sex - allowing us to talk about how the factor of interest (in isolation) is impacting survival, adjusted for other covariates. Can you please verify if the way it is done in the ggadjustedcuve() is different than the one referenced on the website? If I am wrong, please forgive me but I was speaking to one of my superiors and he pointed out that the way it is currently calculated may be just the "Average" curve and not "adjusted" curve.

Thanks alot

**Comment by pbiecek** [comment]
Thanks, will check this on Monday

**Comment by pbiecek** [comment]
IMHO this issue is solved with https://github.com/kassambara/survminer/pull/233

**Comment by kassambara** [comment]
I'm amazed by the update you've done in the `ggadjustedcurves()` function, thank you very much.

I think that we can close this issue.

**Comment by brant01** [comment]
First, thank you for this incredibly valuable package. I have used it extensively. I have having a problem with ggcoxadjustedcurves with large data sets. I can run the coxph, but when I try to make an adjusted curve I get an error that data must be a dictionary n = 4000). If I take a random sample of 100 from this dataset and do the same thing, I do not get this error. Is there an issue with inputting large cox models into ggcoxadjusted curves? I have looked extensively and cannot find a similar error report.

Thanks again for this package and all of the work that went into it.
